sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/BlockBlobInputOutputStreamTest.groovy

```diff
@@ -147,30 +148,308 @@ class BlockBlobInputOutputStreamTest extends APISpec {
+        def blobContainerClient = versionedBlobServiceClient.createBlobContainer(containerName)
+        def blobClient = blobContainerClient.getBlobClient(generateBlobName())
+        blobClient.upload(new ByteArrayInputStream(randomBytes), length, true)
+        def properties = blobClient.getProperties()
+
+        when:
+        def inputStream = blobClient.openInputStream(new BlobInputStreamOptions().setConsistentReadControl(ConsistentReadControl.ETAG)
+        // User provides eTag to use
+            .setRequestConditions(new BlobRequestConditions().setIfMatch(properties.getETag())))
+        def outputStream = new ByteArrayOutputStream()
+
+        then: "Successful read"
+        def b
+        while ((b = inputStream.read()) != -1) {
+            outputStream.write(b)
+        }
+
+        byte[] randomBytes1 = outputStream.toByteArray()
+        assert randomBytes1 == randomBytes
+    }
+
+    def "IS consistent read control etag user provides version and etag"() {
+        setup:
+        int length = Constants.KB
+        byte[] randomBytes = getRandomByteArray(length)
+        def blobContainerClient = versionedBlobServiceClient.createBlobContainer(containerName)
+        def blobClient = blobContainerClient.getBlobClient(generateBlobName())
+        blobClient.upload(new ByteArrayInputStream(randomBytes), length, true)
+        def properties = blobClient.getProperties()
+
+        when:
+        // User provides version client
+        def inputStream = blobClient.getVersionClient(properties.getVersionId()).openInputStream(new BlobInputStreamOptions().setConsistentReadControl(ConsistentReadControl.ETAG)
+        // User provides eTag to use
+            .setRequestConditions(new BlobRequestConditions().setIfMatch(properties.getETag())))
+        def outputStream = new ByteArrayOutputStream()
+
+        then: "Successful read"
+        def b
+        while ((b = inputStream.read()) != -1) {
+            outputStream.write(b)
+        }
+
+        byte[] randomBytes1 = outputStream.toByteArray()
+        assert randomBytes1 == randomBytes
+    }
+
+    def "IS consistent read control etag user provides version client chooses etag"() {
+        setup:
+        int length = Constants.KB
+        byte[] randomBytes = getRandomByteArray(length)
+        def blobContainerClient = versionedBlobServiceClient.createBlobContainer(containerName)
+        def blobClient = blobContainerClient.getBlobClient(generateBlobName())
+        blobClient.upload(new ByteArrayInputStream(randomBytes), length, true)
+        def properties = blobClient.getProperties()
+
+        when:
+        // User provides version client
+        def inputStream = blobClient.getVersionClient(properties.getVersionId()).openInputStream(new BlobInputStreamOptions().setConsistentReadControl(ConsistentReadControl.ETAG))
+        def outputStream = new ByteArrayOutputStream()
+        // When a versioned client is used it should still succeed if the blob has been modified
```

Comment:
Is this true if the specify etag? Or does the etag on the version also get locked so they'll always match?

Reply:
Since the user specifies the version client - we do a getProperties on that version so we get back the etag for that version so it works right. Does that make sense?

If they specify the etag and it is correct - it'll work. If its wrong it will fail as usual and the service will handle that